airflow/ui/src/pages/Dashboard/HistoricalMetrics/MetricsBadge.tsx

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Badge } from "@chakra-ui/react";
+
+type MetricBadgeProps = {
+  readonly color: string;
```

Comment:
We need a backgroundColor and a color param. 

We're getting a lot of bad text colors:
<img width="68" alt="Screenshot 2024-11-12 at 9 12 54 AM" src="https://github.com/user-attachments/assets/e5873d97-c0a3-4a53-b8b0-d0dcbd4b5d9a">

Reply:
Actually, let's just have this type extend `BadgeProps` and pass `{...rest}` at the end of the component.

Reply:
Agreed, maybe similar to `stateColors` mapping there could be a `stateTextColors` mapping for both light and dark modes. I tried applying "fg.inverted" as default for `color` which seems to make it better but still not good for "running" state which is of light green.

Reply:
I agree. We had that for task operator colors already.